src/System.Reflection.Metadata.ILDasm/src/Decoder/ILDecoder.cs

```diff
@@ -0,0 +1,582 @@
+        /// <returns>true if is a type definition false if not</returns>
+        public static bool IsTypeDefinition(int token)
+        {
+            return (token >> 24) == 0x02;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a user string token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a user string false if not</returns>
+        public static bool IsUserString(int token)
+        {
+            return (token >> 24) == 0x70;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a member reference token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a member reference false if not</returns>
+        public static bool IsMemberReference(int token)
+        {
+            return (token >> 24) == 0x0a;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a method specification token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a method specification false if not</returns>
+        public static bool IsMethodSpecification(int token)
+        {
+            return (token >> 24) == 0x2b;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a method definition token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a method definition false if not</returns>
+        public static bool IsMethodDefinition(int token)
+        {
+            return (token >> 24) == 0x06;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a field definition token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a field definition false if not</returns>
+        public static bool IsFieldDefinition(int token)
+        {
+            return (token >> 24) == 0x04;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a type specification token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a type specification false if not</returns>
+        public static bool IsTypeSpecification(int token)
+        {
+            return (token >> 24) == 0x1b;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a standalone signature token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a standalone signature false if not</returns>
+        public static bool IsStandaloneSignature(int token)
+        {
+            return (token >> 24) == 0x11;
+        }
+
+        public static string GetByteArrayString(byte[] array)
```

Comment:
Is the specific format being generated here what's actually required? If any such format could work, callers could just use BitConverter.ToString(array) rather than this method being exposed.